core/controllers/contributor_dashboard.py

```diff
@@ -823,34 +986,79 @@ def _get_client_side_stats(
     stats_dicts = [
         stats.to_dict() for stats in backend_stats
     ]
-    topic_ids = [
-        stats_dict['topic_id']
-        for stats_dict in stats_dicts
-    ]
+    topic_ids = []
+    for stats_dict in stats_dicts:
+        # Here 'topic_id' can never be None because stats_dict is a dictionary
+        # representation of stats object that is fetched from datastore and
+        # before storing stats object in the datastore 'topic_id' is already
+        # assigned.
+        assert stats_dict['topic_id'] is not None
+        topic_ids.append(stats_dict['topic_id'])
     topic_summaries = topic_fetchers.get_multi_topic_summaries(topic_ids)
     topic_name_by_topic_id = {
         topic_summary.id: topic_summary.name
         for topic_summary in topic_summaries if topic_summary is not None
     }
     for stats_dict in stats_dicts:
-        stats_dict['topic_name'] = topic_name_by_topic_id.get(
+        # Here 'topic_id' can never be None because stats_dict is a dictionary
+        # representation of stats object that is fetched from datastore and
+        # before storing stats object in the datastore 'topic_id' is already
+        # assigned.
+        assert stats_dict['topic_id'] is not None
+        # Here we use MyPy ignore because 'stats_dicts' is of union type
```

Comment:
Should the comment refer to stats_dict?

If you are going this route then I think you need a backend test or some check to confirm that each possible element of AllowedStatsTypes has a topic_name field. Though I'm a bit confused about why it's OK to read the topic_id field but not topic_name,

EDIT: by "indexing" do you mean writing / adding a field? Ditto below.

Reply:
> Should the comment refer to stats_dict?

Yes, the comment updated!

> If you are going this route then I think you need a backend test or some check to confirm that each possible element of AllowedStatsTypes has a topic_name field.

topic_name is always going to exist because we are defining topic_name for every dict https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/contributor_dashboard.py#L836 
if topic_id does not exist then topic_name is populated with `UNKNOWN` value. 

> Though I'm a bit confused about why it's OK to read the topic_id field but not topic_name,

Sorry, but can you please explain a bit more about this part?

Reply:
to_frontend_dict() method is added! 
The comment is also updated!